datasource_connect_test.go

```diff
@@ -108,20 +119,21 @@ func Test_getDBConnectionFromQuery(t *testing.T) {
 }
 
 func Test_Dispose(t *testing.T) {
-	t.Run("it should not delete connections", func(t *testing.T) {
-		conn := &Connector{}
-
+	t.Run("it should close connections", func(t *testing.T) {
+		db := sql.OpenDB(fakeSQLConnector{})
+		d := &fakeDriver{openDBfn: func(msg json.RawMessage) (*sql.DB, error) { return db, nil }}
+		conn := &Connector{driver: d}
 		ds := &SQLDatasource{connector: conn}
-		conn.connections.Store(defaultKey("uid1"), dbConnection{})
-		conn.connections.Store("foo", dbConnection{})
+		conn.connections.Store(defaultKey("uid1"), dbConnection{db: db})
+		conn.connections.Store("foo", dbConnection{db: db})
 		ds.Dispose()
 		count := 0
 		conn.connections.Range(func(key, value interface{}) bool {
 			count++
 			return true
 		})
-		if count != 2 {
-			t.Errorf("missing connections")
+		if count != 0 {
```

Comment:
Do we know why this was originally checking that we don't delete connections?

Reply:
I don't! Except that, of course, it originally didn't. I suppose it's possible that existing datasources are relying on that behavior, if their connectors reuse connections? But I feel like that would be a bit weird, and it doesn't look like any of ours do that.

You could argue that this should be a new major version, but to me this feels more like a bugfix than a backwards incompatible change, even if the bug was originally intentional. If that makes sense.

Reply:
(Fwiw I looked over the datasources - grafana and non-grafana - using `sqlds.NewDatasource` and it doesn't look like any of them reuse connections this way.)